File: src/main/java/com/leggo/cooperativa/infrastructure/repositories/mongodb/ProductEntity.java
package com.leggo.cooperativa.infrastructure.repositories.mongodb;

import lombok.Builder;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.Map;

@Document(collection = "product") @Builder @ToString
public class ProductEntity
{
    @Id
    private String productId;
    private String name;
    private double kilogramsPerHectare;
    private MarketRateEntity marketRates;

    @Builder @ToString
    public static class MarketRateEntity
    {
        private Map<LocalDate, BigDecimal> pricesByDay;
        private BigDecimal initialPricePerKilogram;
    }
}

File: src/main/java/com/leggo/cooperativa/infrastructure/repositories/mongodb/ProductEntity.kt
package com.leggo.cooperativa.infrastructure.repositories.mongodb

import java.math.BigDecimal
import java.time.LocalDate
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "product")
data class ProductEntity
(
    @Id
    val productId: String,
    val name: String,
    val kilogramsPerHectare: Double,
    val marketRates: MarketRateEntity,
    val productType: String
)

data class MarketRateEntity
(
    val pricesByDay: Map<LocalDate, BigDecimal>,
    val initialPricePerKilogram: BigDecimal,
)
